backend/src/main/java/planit/people/preparation/Entities/Entity_PresetAvailability.java

```diff
@@ -0,0 +1,189 @@
+     * @param start_available_time   availability start time of a day (since result returned by a query is in Util.Date format, the constructor will convert the value into sql.Time)
+     * @param end_available_time     availability end time of a day (since result returned by a query is in Util.Date format, the constructor will convert the value into sql.Time)
+     */
+    public Entity_PresetAvailability(Long id_preset_availability, Entity_EventPreset entity_EventPreset, WeekDays day, Boolean day_off, java.util.Date start_available_time, java.util.Date end_available_time) {
+        this.id_preset_availability = id_preset_availability;
+        this.entity_EventPreset = entity_EventPreset;
+        this.day = day;
+        this.start_available_time = start_available_time == null ? null : new Time(start_available_time.getTime());
+        this.end_available_time = end_available_time == null ? null : new Time(end_available_time.getTime());
+        this.day_off = day_off;
+    }
+
+    /**
+     * Constructor used in the process of creating a new record.
+     *
+     * @param id_preset_availability id of the preset availability
+     * @param day                    day of a record
+     * @param day_off                day off
+     * @param start_available_time   availability start time of a day
+     * @param end_available_time     availability end time of a day
+     */
+    public Entity_PresetAvailability(Long id_preset_availability, WeekDays day, Boolean day_off, Time start_available_time, Time end_available_time) {
+        this.id_preset_availability = id_preset_availability;
+        this.day = day;
+        this.start_available_time = start_available_time;
+        this.end_available_time = end_available_time;
+        this.day_off = day_off;
+    }
+
+    /**
+     * Constructor used in creating test methods
+     *
+     * @param day                  day of a record
+     * @param day_off              day off
+     * @param start_available_time availability start time of a day
+     * @param end_available_time   availability end time of a day
+     */
+    public Entity_PresetAvailability(WeekDays day, Time start_available_time, Time end_available_time, Boolean day_off) {
+        this.day = day;
+        this.start_available_time = start_available_time;
+        this.end_available_time = end_available_time;
+        this.day_off = day_off;
+    }
+
+    public Long getId_preset_availability() {
+        return id_preset_availability;
+    }
+
+    public void setId_preset_availability(Long id_preset_availability) {
+        this.id_preset_availability = id_preset_availability;
+    }
+
+    public Boolean getDay_off() {
+
+        return day_off;
+    }
+
+    public void setDay_off(Boolean day_off) {
+        this.day_off = day_off;
+    }
+
+    public Time getEnd_available_time() {
+        return end_available_time;
+    }
+
+    public void setEnd_available_time(Time end_available_time) {
```

Comment:
why not `setEndAvailableTime`?

Reply:
Auto generated setter because the field name is the_user